src/middlewares/validation.ts

```diff
@@ -779,6 +779,39 @@ const isUserWishlistExistById = async (
   }
 };
 
+const isProductOrdered = async (
+  req: ExtendRequest,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const carts = await cartRepositories.getCompletedCartsByUserId(req.user.id);

```

Comment:
Change the logic and then start by getting a **`cartProducts`** by **`productId`** then include **`Cart`** and then check if the **`cart.status === 'completed'`** and finally check with this condition **`cart.userId !== req.user.id`**. Once everything is good proceed to the next logic.